src/cli/keepassxc-cli.cpp

```diff
@@ -33,6 +39,131 @@
 #include <sanitizer/lsan_interface.h>
 #endif
 
+#if defined(USE_READLINE)
+#include <readline/history.h>
+#include <readline/readline.h>
+#endif
+
+class LineReader
+{
+public:
+    virtual ~LineReader() = default;
+    virtual QString readLine(QString prompt) = 0;
+    virtual bool isFinished() = 0;
+};
+
+class SimpleLineReader : public LineReader
+{
+public:
+    SimpleLineReader()
+        : inStream(stdin, QIODevice::ReadOnly)
+        , outStream(stdout, QIODevice::WriteOnly)
+        , finished(false)
+    {
+    }
+
+    QString readLine(QString prompt) override
+    {
+        outStream << prompt;
+        outStream.flush();
+        QString result = inStream.readLine();
+        if (result.isNull()) {
+            finished = true;
+        }
+        return result;
+    }
+
+    bool isFinished() override
+    {
+        return finished;
+    }
+
+private:
+    TextStream inStream;
+    TextStream outStream;
+    bool finished;
+};
+
+#if defined(USE_READLINE)
+class ReadlineLineReader : public LineReader
+{
+public:
+    ReadlineLineReader()
+        : finished(false)
+    {
+    }
+
+    QString readLine(QString prompt) override
+    {
+        char* result = readline(prompt.toStdString().c_str());
+        if (!result) {
+            finished = true;
+            return {};
+        }
+        add_history(result);
+        QString qstr(result);
+        free(result);
+        return qstr;
+    }
+
+    bool isFinished() override
+    {
+        return finished;
+    }
+
+private:
+    bool finished;
+};
+#endif
+
+void enterInteractiveMode(const QStringList& arguments)
+{
+
+    Open o;
+    QStringList openArgs(arguments);
+    openArgs.removeFirst();
+    o.execute(openArgs);
+
+    QScopedPointer<LineReader> reader;
+#if defined(USE_READLINE)
+    reader.reset(new ReadlineLineReader());
+#else
+    reader.reset(new SimpleLineReader());
+#endif
+
+    QSharedPointer<Database> currentDatabase(o.currentDatabase);
+
+    QString command;
+    while (true) {
+        TextStream errorTextStream(Utils::STDERR, QIODevice::WriteOnly);
+
+        QString prompt;
+        if (currentDatabase) {
+            prompt += QDir::current().relativeFilePath(currentDatabase->filePath());
```

Comment:
Instead of relative path you should show the database name `currentDatabase->metadata()->name()` or if that is empty `QFileInfo(currentDatabase->filePath()).fileName()`. You should cache this information outside the while loop and only change it if the database changes.